src/autobridgecalculator.js

```diff
@@ -46,7 +46,27 @@ AutobridgeCalculator.prototype.calculate = function() {
 
   this.clearOwnerFundsLeftover();
 
+  this._calculateInternal(legOnePointer, legTwoPointer, offersAutobridged,
+    callback);
+};
+
+AutobridgeCalculator.prototype._calculateInternal = function(
+  legOnePointer, legTwoPointer, offersAutobridged, callback = function() {}
+) {
+
+  const startTime = Date.now();
+
   while (this.legOneOffers[legOnePointer] && this.legTwoOffers[legTwoPointer]) {
+    const lasted = (Date.now() - startTime);
```

Comment:
Ok, I think you should add a comment like "manually implement cooperative multitasking that yields after 30ms of execution and restarts after 33ms". Also, wouldn't it still work if it was `setTimeout(func, 0)`? With the 33ms delay it will be wasting a lot of time if you're only running with one orderbook.